src/client/chat.js

```diff
@@ -346,6 +346,8 @@ module.exports = function (client, options) {
         salt: options.salt,
         argumentSignatures: signaturesForCommand(command, options.timestamp, options.salt),
         signedPreview: options.didPreview,
+        messageCount: 0,
+        acknowledged: Buffer.alloc(3),
```

Comment:
This code block only runs in versions without chat sessions (1.19 through 1.19.2), so no need to add fields for later versions

Reply:
Do chat commands still use chat sessions / chat_message packets?

I have this pull request sending chat commands to nmp for building the chat packet: https://github.com/PrismarineJS/mineflayer/pull/3110. But then that changes chat commands from being sent as chat_command to chat_message because of useChatSessions. If I remove the useChatSessions check for chat_commands, then these fields are required.

It seems that if I remove `&& !mcData.supportFeature('useChatSessions')` it makes the pull request work

Reply:
You're gonna want to create a separate block for versions that use chat sessions. Currently you are not actually providing the message acknowledgements, which will break things on servers enforcing secure chat

Reply:
This is copying the implementation change inside mineflayer that should have been in nmp, so there wouldn't be a change from the status quo on implementation. But yes in a later PR that can be done after the command signing #1067 change is handled in mineflayer

Reply:
For acknowledgements we can definitely just copy what we do for regular chat signing

Reply:
I created a new block mixing the two. So it sends a chat_command, but does the acknoledgement stuff. I didn't know what to put for messageCount however. I created a 1.19.3 server, and the bot gets kicked on master and on this version.